OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -115,7 +115,7 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 				var movement = self.Trait<IPositionable>();
 
 				var minefield = GetMinefieldCells(minefieldStart, cell, Info.MinefieldDepth)
-					.Where(c => movement.CanEnterCell(c, null, BlockedByActor.Immovable) || !self.Owner.Shroud.IsVisible(c))
+					.Where(c => movement.CanEnterCell(c, null, BlockedByActor.Immovable))
```

Comment:
This doesn't look correct. Won't people now be able to find out which cells are passable beneath shroud? This should probably just have a Map.Contains instead.